bindings/ffi/README.md

```
@@ -0,0 +1,86 @@
+# IMPORTANT: Free the string that was allocated by Rust
+gatos_lib.gatos_ffi_free_string(message_ptr)
+```
+
+## Safety Considerations
+
+Working with an FFI boundary requires careful attention to memory management and safety.
+
+- **Ownership:** Any string or complex object returned by the GATOS FFI was allocated by Rust and its ownership is transferred to the caller. The caller is **required** to return control of the object to Rust for safe deallocation.
+- **Memory Management:** All strings returned by this library **must** be freed by calling the provided `gatos_ffi_free_string` function. Failure to do so will result in memory leaks.
+- **Null Pointers:** Do not pass `NULL` pointers to any function unless the function's documentation explicitly states that it is allowed.
+- **Thread Safety:** The thread safety of the FFI functions is not guaranteed unless explicitly stated. Avoid calling FFI functions from multiple threads without proper synchronization.
```

markdownlint-cli2 (0.18.1)

Files should end with a single newline character

(MD047, single-trailing-newline)